crates/syntax/src/ast/make.rs

```diff
@@ -520,6 +520,19 @@ pub fn literal_pat(lit: &str) -> ast::LiteralPat {
     }
 }
 
+pub fn slice_pat(pats: impl IntoIterator<Item = ast::Pat>) -> ast::SlicePat {
+    let mut count: usize = 0;
+    let mut pats_str = pats.into_iter().inspect(|_| count += 1).join(", ");
+    if count == 1 {
+        pats_str.push(',');
+    }
```

Comment:
This part isn't needed here sincc a slice pat is always a slice pat no matter the commas (unlike tuple patterns where `(foo)` is a parenthesized pattern vs `(foo,)` which is a 1-ary tuple pattern)

Reply:
done, thanks for your tips, and test case had been added:
- [match \[true\]](https://github.com/rust-lang/rust-analyzer/pull/13978/files#diff-255170f3d2563a96c7ad33c081dee2a9359c14d3e1ddae275bc5a95b9ea7ac58R620)
- [match \[true, \]](https://github.com/rust-lang/rust-analyzer/pull/13978/files#diff-255170f3d2563a96c7ad33c081dee2a9359c14d3e1ddae275bc5a95b9ea7ac58R638)